freshmaker/handlers/botas/botas_shipped_advisory.py

```diff
@@ -626,6 +626,37 @@ def _create_original_to_rebuilt_nvrs_map(self):
                         and block_build_nvr["version"] == parsed_build_nvr["version"]  # noqa: W503
                     ):
                         nvrs_mapping[block_build] = build_nvr
+
+        if self.event.get("autocomplete_field", False): #TODO implement "autocomplete" field of event
+            self._add_parent_nvrs(nvrs_mapping)
+
+        return nvrs_mapping
+
+    def _add_parent_nvrs(nvrs_mapping: dict[str, str]) -> dict[str, str]:
+        """Adds chain of parent nvrs of freshmaker rebuilds to the current nvr mapping
+        
+        This function searches the parent builds of the nvrs in the nvr mapping of the current
+        event and adds them to the mapping, in case they are previous freshmaker rebuilds.
+        
+        This can be used to trigger rebuilds of bundles that lost the "link" to a latest version
+        of an image (due to problems with a former rebuild process, for example) in an automatic
+        fashion, without requiring manual overriding.
+        """
+        nvr_mappings_to_add = {}
+        for nvr in nvrs_mapping.keys():  # outer loop on original nvrs_mapping keys
+            child_nvr = nvr
+            parent_nvr = True
+            while parent_nvr:  # inner loop on chain of parents of each nvr 
+                parent_nvr = None
+                stmt = select(ArtifactBuild).filter_by(rebuilt_nvr=child_nvr)
+                query_result = db.session.execute(stmt).first()
+                if query_result:
+                    # query_result is a tuple with the artifact build object on its first position
+                    parent_nvr = query_result[0].original_nvr
+                    nvr_mappings_to_add.update({parent_nvr: nvrs_mapping[nvr]})
+                    child_nvr = parent_nvr
+        
+        nvrs_mapping.update(nvr_mappings_to_add)
```

Comment:
Please also add logging (about which NVRs have been added) to help with troubleshooting.

Reply:
I created two log calls, one in the `while` loop in `_add_previous_rebuilds` and another in the `if` clause in `_create_original_to_rebuilt_nvrs_map`. Please let me know if you think that is excessive, if there is information missing, or anything that could improve the logging!